packages/api/src/controllers/animalController.js

```diff
@@ -0,0 +1,115 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { Model, transaction } from 'objection';
+import AnimalModel from '../models/animalModel.js';
+import baseController from './baseController.js';
+import DefaultAnimalBreedModel from '../models/defaultAnimalBreedModel.js';
+import CustomAnimalBreedModel from '../models/customAnimalBreedModel.js';
+
+const animalController = {
+  getFarmAnimals() {
+    return async (req, res) => {
+      try {
+        const { farm_id } = req.headers;
+        const rows = await AnimalModel.query().where({ farm_id });
```

Comment:
Probably not, but I wasn't sure since we're not filtering deleted records out for types and breeds. @kathyavini did you have any thoughts on this, or should we actually add that condition on all controllers?

Reply:
Sorry I read this in the morning and then forgot! Yes, all controllers should probably have it 🙂 

I think we still haven't definitively decided whether we are excluding deleted entities here, or again overriding the hidden properties to filter on the frontend -- and maybe RTK Query will impact on which of those we should do? -- but `.whereNotDeleted()` is a good default to start with and we can always adjust later.